### DISCUSSION TITLE
How can I check if the value is null or valid value? Required or present rule doesn't fit in this case, because it fails on null or cancels further checks.

### DISCUSSION THREAD

**Discussion by FragsterAt** [laravel/framework#52266] [Q&A]
```php
dump(Validator::make(["foo" => ""], ["foo" => "nullable|present|date"])->passes());
dump(Validator::make(["foo" => ""], ["foo" => "nullable|string|min:3"])->passes());
```

etc

Required fails on null value, present succeed on any values but other checks are not executed on empty strings.

I want to pass data direct to database with nullable columns and now I have errors like `Invalid datetime format: 1292 Incorrect date value: '' for column` instead of validation errors

There is closed issues #52262 #50626 about this problem

**Comment by ghost** [comment]
Custom validation rule or 'sometimes|nullable|string|min:3'. For date you need to pass a format.

**Reply by FragsterAt** [reply]
> Custom validation rule or 'sometimes|nullable|string|min:3'. For date you need to pass a format.

```php
  dump(Validator::make(["foo" => ""], ["foo" => "sometimes|nullable|string|min:3"])->passes());
```

passes but must fail

reason is in https://github.com/laravel/framework/blob/b18f33340ca5a691132bccb3ad5de59c7fbdfdd9/src/Illuminate/Validation/Validator.php#L777 that skip all rules on empty strings
